src/github_client.rs

```diff
@@ -30,18 +40,23 @@ impl GithubClient {
         }
     }
 
-    /// Gets all notifications since a timestamp even if they've been read
-    async fn get_notifications(&self, since: DateTime<Local>) -> Result<Response, Error> {
+    /// Gets all notifications since a timestamp, even if they've been read.
+    async fn get_notifications(
+        &self,
+        since: DateTime<Local>,
+    ) -> Result<Vec<GithubNotification>, Error> {
         self.client
             .get(GITHUB_API_URL)
             .basic_auth(&self.username, Some(&self.token))
             .query(&[("since", since.to_rfc3339()), ("all", String::from("true"))])
             .send()
+            .await?
+            .json::<Vec<GithubNotification>>()
```

Comment:
I was initially thinking about trying to make a more generic GitHub api request function but I suspect that:
A) that unnecessarily decouples the deseirlaization logic from the request logic
B) if we added more logic for handling different queries then we'd just be reinventing the requests wheel and it'd be a thing enough wrapper that it'd be kind of useless.

So this seems totally fine to squash together.